cluster/addons/fluentd-gcp/scaler-policy.yaml

```diff
@@ -1,12 +1,13 @@
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   name: scalingpolicies.scalingpolicy.kope.io
   labels:
     addonmanager.kubernetes.io/mode: Reconcile
 spec:
   group: scalingpolicy.kope.io
-  version: v1alpha1
+  versions:
+    - name: v1alpha1
```

Comment:
this will not apply successfully, right? the `schema` and `storage` attributes are required in v1

Reply:
also, when transitioning from v1beta1 to v1, I'd recommend setting spec.preserveUnknownFields to `false` explicitly, so you don't inherit the default `true` value from the previously persisted v1beta1 CRD

Reply:
Hi Jordan,
It seems that `schema` is not required, but `storage` and `served` are required. I have prepared #101238 to fix it. Related to preserveUnknownFields, it seems that this field is [deprecated](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#customresourcedefinitionspec-v1-apiextensions-k8s-io) and instead corresponding field should be set in schema.

Reply:
> Related to preserveUnknownFields, it seems that this field is deprecated

when updating from a beta version that left it populated with true, you should explicitly set it to false in the v1 version

Reply:
Hm, interesting. When I have upgraded CRD from v1beta1 everything worked without `scheme` field. But when I try to create it from scratch, this field is required. Is it expected, or should I file a bug?

Reply:
> Is it expected, or should I file a bug?

That's expected. Updates of previously created objects that don't meet v1 requirements have to be tolerated for backwards compatibility, but all v1 requirements are enforced when creating new CRDs via the v1 API